### PR TITLE
Add pull request url to PullRequestInfo

### PR DESCRIPTION
**Pull request by omerzi**
- [x] All [tests](https://github.com/jfrog/froggit-go/actions/workflows/test.yml) passed. If this feature is not already covered by the tests, I added new tests.
- [x] I used `go fmt ./...` for formatting the code before submitting the pull request.
- [x] This feature is included on all supported VCS providers - GitHub, Bitbucket cloud, Bitbucket server, and GitLab.

---

### CODE REVIEW

File: vcsclient/github.go [hunk 1]
```diff
@@ -336,27 +336,53 @@ func (client *GitHubClient) GetPullRequestByID(ctx context.Context, owner, repos
...
-	err = errors.Join(err1, err2)
-	if err != nil {
-		return PullRequestInfo{}, err
+	return mapGitHubPullRequestToPullRequestInfo(pullRequest, false)
+}
+
+func mapGitHubPullRequestToPullRequestInfo(ghPullRequest *github.PullRequest, withBody bool) (PullRequestInfo, error) {
+	var sourceBranch, targetBranch string
+	var err1, err2 error
+	if ghPullRequest != nil && ghPullRequest.Head != nil && ghPullRequest.Base != nil {
```
Comment (sverdlov93) [inline vcsclient/github.go:345]: why?

File: vcsclient/github.go [hunk 2]
```diff
@@ -336,27 +336,53 @@ func (client *GitHubClient) GetPullRequestByID(ctx context.Context, owner, repos
...
+		sourceBranch, err1 = extractBranchFromLabel(vcsutils.DefaultIfNotNil(ghPullRequest.Head.Label))
+		targetBranch, err2 = extractBranchFromLabel(vcsutils.DefaultIfNotNil(ghPullRequest.Base.Label))
+		err := errors.Join(err1, err2)
+		if err != nil {
+			return PullRequestInfo{}, err
+		}
+	}
+
+	var sourceRepoName, sourceRepoOwner string
+	if ghPullRequest.Head.Repo != nil && ghPullRequest.Head.Repo.Owner != nil {
```
Comment (sverdlov93) [inline vcsclient/github.go:355]: why?
what happens if its not true?

**Comment by sverdlov93** [comment]
PULL_REQUEST_TEMPLATE.md >> 
This feature is included on all supported VCS providers - GitHub, Bitbucket cloud, Bitbucket server, and GitLab.

Azure is missing

File: vcsclient/bitbucketserver.go [hunk 3]
```diff
@@ -345,30 +345,12 @@ func (client *BitbucketServerClient) getOpenPullRequests(ctx context.Context, ow
 			return nil, err
```
Comment (sverdlov93) [inline vcsclient/bitbucketserver.go:345]: what about bitbucket cloud?

File: vcsclient/github.go (hunk 1 above)
Comment (omerzi) [reply vcsclient/github.go:345]: Head and Base are pointers, could have a nil values

File: vcsclient/github.go (hunk 2 above)
Comment (omerzi) [reply vcsclient/github.go:355]: Then we'll send empty values, do you think we need to throw an error in these cases? @sverdlov93

File: vcsclient/github.go (hunk 1 above)
Comment (omerzi) [reply vcsclient/github.go:345]: @sverdlov93

File: vcsclient/bitbucketserver.go (hunk 3 above)
Comment (omerzi) [reply vcsclient/bitbucketserver.go:345]: We are usually not implementing Bitbucket Cloud for new APIs as we currently don't need it.

**Pull request merged**